src/main/kotlin/nl/info/zac/app/zaak/model/BetrokkeneIdentificatie.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2025 INFO.nl
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package nl.info.zac.app.zaak.model
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import nl.info.zac.app.klant.model.klant.IdentificatieType
+import nl.info.zac.util.AllOpen
+import nl.info.zac.util.NoArgConstructor
+import org.jetbrains.annotations.NotNull
+import kotlin.reflect.KClass
+
+@AllOpen
+@NoArgConstructor
+@ValidBetrokkeneIdentificatie
+data class BetrokkeneIdentificatie(
+    @field:NotNull
+    var type: IdentificatieType,
+    var bsnNummer: String? = null,
+    var kvkNummer: String? = null,
+    var vestigingsnummer: String? = null,
+    var rsinNummer: String? = null
+)
+
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [BetrokkeneIdentificatieValidator::class])
+annotation class ValidBetrokkeneIdentificatie(
+    val message: String = "Invalid BetrokkeneIdentificatie",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<out Payload>> = []
+)
+
+class BetrokkeneIdentificatieValidator : ConstraintValidator<ValidBetrokkeneIdentificatie, BetrokkeneIdentificatie> {
+    override fun isValid(value: BetrokkeneIdentificatie?, context: ConstraintValidatorContext): Boolean {
```

Comment:
Do you need to allow for a null value? I would say: you should never even be able to call this function if you have a null value because why would you? You need to do the null check before calling this function (pretty much any such function really). Or is this required by the ConstraintValidator interface?